packages/application-shell/src/components/authenticated/helpers.ts

```diff
@@ -0,0 +1,34 @@
+import type { ApplicationWindow } from '@commercetools-frontend/constants';
+
+import { OIDC_CLAIMS } from '../../constants';
+
+declare let window: ApplicationWindow;
+
+export const buildOidcScope = (options: {
+  projectKey?: string | null;
+}): string => {
+  const projectClaims = [];
+  if (options.projectKey) {
+    projectClaims.push(`${OIDC_CLAIMS.PROJECT_KEY}:${options.projectKey}`);
+    projectClaims.push(
+      ...(window.app.__DEVELOPMENT__?.permissions?.view ?? []).map(
+        (scope) => `${OIDC_CLAIMS.VIEW}:${scope}`
+      )
+    );
+    projectClaims.push(
+      ...(window.app.__DEVELOPMENT__?.permissions?.manage ?? []).map(
+        (scope) => `${OIDC_CLAIMS.MANAGE}:${scope}`
+      )
+    );
+  }
+  const teamId = window.app.__DEVELOPMENT__?.teamId;
+  if (teamId) {
+    projectClaims.push(`${OIDC_CLAIMS.TEAM_ID}:${teamId}`);
+  }
+  return [
+    // This is required as per OIDC spec.
+    OIDC_CLAIMS.OPEN_ID,
+    // Custom claims
+    ...projectClaims,
+  ].join(' ');
+};
```

Comment:
How would we test store scopes with this? Is it intended or a possible scenario?

Reply:
I think it can be defined as a normal scope.

Reply:
I scope is `manage_orders:{projectKey}:{storeKey}`. So if I add `view.[0] = 'orders:storeA` would that work?

Reply:
I double checked. It's not supported at the moment.

I think we can fix this in the backend.

So you would still pass the oauth scopes as `view_products` or `view_orders:storeA`.